### PR TITLE
docs: add helm-version and yaml-version shortcode

### PR DESCRIPTION
**Pull request by zirain**
Thi will make it simpler to manage the version of helm chart and yaml file in doc site.

### CODE REVIEW

File: site/layouts/shortcodes/helm-version.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{{- $pagePrefix := (index (split $.Page.File.Dir "/") 0) -}}
+{{- with (eq $pagePrefix "latest") -}}
+{{- "v0.0.0-latest" -}}
+{{- end -}}
+{{- with (strings.HasPrefix $pagePrefix "v1.1") -}}
+{{- "v1.1.0" -}}
```
Comment (zirain) [inline site/layouts/shortcodes/helm-version.html:6]: need update this when releasing a minor patch.

File: site/layouts/shortcodes/helm-version.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{{- $pagePrefix := (index (split $.Page.File.Dir "/") 0) -}}
+{{- with (eq $pagePrefix "latest") -}}
+{{- "v0.0.0-latest" -}}
+{{- end -}}
+{{- with (strings.HasPrefix $pagePrefix "v1.1") -}}
```
Comment (zirain) [inline site/layouts/shortcodes/helm-version.html:5]: need update this when releasing a major patch.

**Comment by zhaohuabing** [comment]
@zirain Why do we need this?

**Comment by zirain** [comment]
> @zirain Why do we need this?

better to have this.

File: site/layouts/shortcodes/helm-version.html (hunk 1 above)
Comment (zhaohuabing) [reply site/layouts/shortcodes/helm-version.html:6]: Do we also need to update the release guide?

Comment (zirain) [reply site/layouts/shortcodes/helm-version.html:6]: Absolutely, there were a lot of mistakes on these things.

**Review by shawnh2 — DISMISSED** [review]
LGTM, thanks for optimizing this!

**Review by zhaohuabing — APPROVED** [review]
LGTM Thanks!

File: site/content/en/contributions/RELEASING.md [hunk 3]
```diff
@@ -97,34 +97,56 @@ export GITHUB_REMOTE=origin
    1. Create the release notes. Reference previous [release notes][] for additional details. __Note:__  The release
       notes should be an accumulation of the release candidate release notes and any changes since the release
       candidate.
-   2. Create a release announcement. Refer to [PR #635] as an example release announcement.
-   3. Include the release in the compatibility matrix. Refer to [PR #1002] as an example.
-   4. Generate the versioned release docs:
+   1. Create a release announcement. Refer to [PR #635] as an example release announcement.
```
Comment (shawnh2) [inline site/content/en/contributions/RELEASING.md:100]: nit: those numbers are not correct

Comment (zirain) [reply site/content/en/contributions/RELEASING.md:100]: https://github.com/envoyproxy/gateway/pull/635
https://github.com/envoyproxy/gateway/pull/1002

**Comment by zirain** [comment]
wait for https://github.com/envoyproxy/gateway/pull/3910

**Review by shawnh2 — APPROVED** [review]

**Pull request merged**